pkg/expression/column.go

```diff
@@ -511,9 +511,13 @@ func (col *Column) Hash64(h base.Hasher) {
 
 // Equals implements HashEquals.<1st> interface.
 func (col *Column) Equals(other any) bool {
-	if other == nil {
-		return false
+	// If col itself is nil, check whether other is also nil
+	if col == nil {
```

Comment:
better use this format
```
	op2, ok := other.(*DataSource)
	if !ok {
		return false
	}
	if op == nil {
		return op2 == nil
	}
	if op2 == nil {
		return false
	}
```

and there many other places implements this Equals interface with the same bad way, we should change them too

Reply:
![image](https://github.com/user-attachments/assets/c1cb0672-fdbc-455a-aae7-084f28b82cf2)

you can do it by batch with 2-3, or finish them all, it's up to you, do remember the test case